airflow/jobs/scheduler_job.py

```diff
@@ -795,7 +795,7 @@ def _create_dagruns_for_dags(self, guard, session):
         guard.commit()
         # END: create dagruns
 
-    def _create_dag_runs(self, dag_models: Iterable[DagModel], session: Session) -> None:
+    def _create_dag_runs(self, dag_models: Collection[DagModel], session: Session) -> None:
```

Comment:
This annotation was wrong. `dag_models` is iterated over multiple times in this function, and an iterable does not satisfy that usage (an iterator can only be iterated through once).